### PR TITLE
DPMMA-2825 Fix creating tag duplicates with encoded special characters

### LINKED ISSUE #13922: Duplicate tags created during CSV contact import when special characters are present
### Mautic Version

5.0.x series

### Way of installing

I downloaded a release from https://www.mautic.org/mautic-releases

### PHP version

8.0

### What browsers are you seeing the problem on?

Not relevant

### What happened?

When importing contacts from a CSV file using the Tags option, tags that contain special characters (such as &, ', or ") are being duplicated. This results in multiple instances of the same tag, leading to inconsistencies in the tagging system.

### How can we reproduce this issue?

Step 1: Create a CSV file with contact data to import, for example:
```
email,name
test@example.com,test
```
Step 2: Go to contact import view: Contacts -> Import
Step 3: Select file and click upload
Step 4: Match the columns and add a Tag with special char, for example `R&D`
Step 5: Click import in browser
Step 6: Go to Tags and check if there is a duplicate
image

### Relevant log output

_No response_

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

### PR DESCRIPTION
**Pull request by patrykgruszka**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🟢 
| New feature/enhancement? (use the a.x branch)      | 🔴
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🟢 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #13922

## Description
This PR fixes the issue with tag duplicates caused by different tag name encoding in cases:
- contact import
- campaign change tags action
- point action change tags
- form action change tags

---
### 📋 Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Follow the steps to reproduce from issue #13922. 

This also can be tested with the campaign change tags action, point action change tags, and form action change tags.

### CODE REVIEW

File: app/bundles/PageBundle/Tests/Controller/PublicControllerFunctionalTest.php [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+        // Check the response content to ensure no script is present
+        $content = $this->client->getResponse()->getContent();
+        Assert::assertStringNotContainsString($payload, $content);
+    }
+
+    /**
+     * @return array<string, array<int, string|null>>
+     */
+    public function xssPayloadsProvider(): array
```
Comment (tomekkowalczyk) [inline app/bundles/PageBundle/Tests/Controller/PublicControllerFunctionalTest.php]: Method should be static

**Review by tomekkowalczyk — COMMENTED** [review]
@patrykgruszka Googd job, I left one comment

File: app/bundles/PageBundle/Tests/Controller/PublicControllerFunctionalTest.php (hunk 1 above)
Comment (patrykgruszka) [reply app/bundles/PageBundle/Tests/Controller/PublicControllerFunctionalTest.php]: Changed to static

**Review by tomekkowalczyk — APPROVED** [review]

**Review by shinde-rahul — APPROVED** [review]
The changes looks good to me!

**Review by Bastian2718 — APPROVED** [review]
Duplicate is there.

**Pull request merged**